code/modules/research/anomaly/explosive_compressor.dm

```diff
@@ -128,18 +124,21 @@
 		inserted_core.forceMove(drop_location())
 		inserted_core = null
 		say("Transfer valve resulted in negligible explosive power. Items ejected.")
-		return
+		return FALSE
 	mix.react() // build more pressure
-	var/pressure = mix.return_pressure()
-	var/range = (pressure - TANK_FRAGMENT_PRESSURE) / TANK_FRAGMENT_SCALE
+
+	var/range = (mix.return_pressure() - TANK_FRAGMENT_PRESSURE) / TANK_FRAGMENT_SCALE
+	QDEL_NULL(inserted_bomb) // bomb goes poof
 	if(range < required_radius)
-		inserted_bomb.forceMove(src)
 		say("Resultant detonation failed to produce enough implosive power to compress [inserted_core]. Core ejected.")
-		return
-	QDEL_NULL(inserted_bomb) // bomb goes poof
+		inserted_core.forceMove(drop_location())
+		inserted_core = null
+		return FALSE
```

Comment:
Behavior change?

Reply:
It claimed to return the uncompressed core but never actually did so. Ifix.

Reply:
👍